apps/passport-client/src/dispatch.ts

```diff
@@ -853,6 +853,39 @@ async function doSync(
         state.pcds,
         state.credentialCache
       );
+
+      /**
```

Comment:
This is stylistic feedback, so take or leave as you wish:
I think this functionally makes sense, but stylistically it seems like a lot of complexity to be put specifically in this sync function (which is already pretty complex in other ways).
Could this logic happen inside of SubscriptionManager.pollSubscriptions, perhaps by marking one specific subscription as "poll this first"?
Failing that, I'd suggest this code block would be a good candidate to break into a helper function so the details of its implementation (and the big explanatory comment) don't make it harder to understand the overall purpose of the sync code here.